core-bundle/src/Resources/public/core.js

```diff
@@ -920,16 +920,31 @@ var Backend =
 				return;
 			}
 			var frm = window.frames['simple-modal-iframe'],
-				val = [], ul, inp, field, act, it, i;
+				val = [], ul, inp, field, act, it, i, pickerValue, sIndex;
 			if (frm === undefined) {
 				alert('Could not find the SimpleModal frame');
 				return;
 			}
 			ul = frm.document.getElementById(opt.id);
+			// Load the previous values (#1816)
+			if (pickerValue = ul.get('data-picker-value')) {
+				val = JSON.parse(pickerValue);
+			}
 			inp = ul.getElementsByTagName('input');
 			for (i=0; i<inp.length; i++) {
-				if (inp[i].checked && !inp[i].id.match(/^(check_all_|reset_)/)) {
-					val.push(inp[i].get('value'));
+				if (inp[i].id.match(/^(check_all_|reset_)/)) {
+					continue;
+				}
+				// Add currently selected value, otherwise remove (#1816)
+				sIndex = val.indexOf(inp[i].get('value'));
+				if (inp[i].checked) {
+					if (sIndex < 0) {
+						val.push(inp[i].get('value'));
+					}
+				} else {
+					if (sIndex >= 0) {
+						val.splice(sIndex, 1);
+					}
```

Comment:
```suggestion
				} else if (sIndex >= 0) {
					val.splice(sIndex, 1);
```

Reply:
imho the code is more readable without `else if`. Before:

```js
if (inp[i].checked) {
    if (sIndex < 0) {
        val.push(inp[i].get('value'));
    }
} else {
    if (sIndex >= 0) {
        val.splice(sIndex, 1);
    }
}
```

after:

```js
if (inp[i].checked) {
    if (sIndex < 0) {
        val.push(inp[i].get('value'));
    }
} else if (sIndex >= 0) {
    val.splice(sIndex, 1);
}
````

Even though both the `val.push` and the `val.splice` require two conditions to be executed (`checked && sIndex === -1` and `!checked && sIndex !== -1`), they are now on different indentation levels.

Reply:
> imho the code is more readable without else if

I tend to disagree. Imo both `if`s should also be consolidated.

Reply:
Like this you mean? fe1beb1cde01a8907f51da63d93777c56ebe2b52